parachains/runtimes/collectives/collectives-polkadot/src/lib.rs

```diff
@@ -0,0 +1,833 @@
+	type OperationalFeeMultiplier = OperationalFeeMultiplier;
+}
+
+parameter_types! {
+	// One storage item; key size is 32; value is size 4+4+16+32 bytes = 56 bytes.
+	pub const DepositBase: Balance = deposit(1, 88);
+	// Additional storage item size of 32 bytes.
+	pub const DepositFactor: Balance = deposit(0, 32);
+	pub const MaxSignatories: u16 = 100;
+}
+
+impl pallet_multisig::Config for Runtime {
+	type Event = Event;
+	type Call = Call;
+	type Currency = Balances;
+	type DepositBase = DepositBase;
+	type DepositFactor = DepositFactor;
+	type MaxSignatories = MaxSignatories;
+	type WeightInfo = weights::pallet_multisig::WeightInfo<Runtime>;
+}
+
+impl pallet_utility::Config for Runtime {
+	type Event = Event;
+	type Call = Call;
+	type PalletsOrigin = OriginCaller;
+	type WeightInfo = weights::pallet_utility::WeightInfo<Runtime>;
+}
+
+parameter_types! {
+	// One storage item; key size 32, value size 8; .
+	pub const ProxyDepositBase: Balance = deposit(1, 40);
+	// Additional storage item size of 33 bytes.
+	pub const ProxyDepositFactor: Balance = deposit(0, 33);
+	pub const MaxProxies: u16 = 32;
+	// One storage item; key size 32, value size 16
+	pub const AnnouncementDepositBase: Balance = deposit(1, 48);
+	pub const AnnouncementDepositFactor: Balance = deposit(0, 66);
+	pub const MaxPending: u16 = 32;
+}
+
+/// The type used to represent the kinds of proxying allowed.
+#[derive(
+	Copy,
+	Clone,
+	Eq,
+	PartialEq,
+	Ord,
+	PartialOrd,
+	Encode,
+	Decode,
+	RuntimeDebug,
+	MaxEncodedLen,
+	scale_info::TypeInfo,
+)]
+pub enum ProxyType {
+	/// Fully permissioned proxy. Can execute any call on behalf of _proxied_.
+	Any,
+	/// Can execute any call that does not transfer funds.
+	NonTransfer,
+	/// Proxy with the ability to reject time-delay proxy announcements.
+	CancelProxy,
+	/// Collator selection proxy. Can execute calls related to collator selection mechanism.
+	Collator,
+	/// Alliance proxy. Allows calls related to the Alliance.
+	Alliance,
+}
+impl Default for ProxyType {
+	fn default() -> Self {
+		Self::Any
+	}
+}
+impl InstanceFilter<Call> for ProxyType {
+	fn filter(&self, c: &Call) -> bool {
+		match self {
+			ProxyType::Any => true,
+			ProxyType::NonTransfer => !matches!(c, Call::Balances { .. }),
+			ProxyType::CancelProxy => matches!(
+				c,
+				Call::Proxy(pallet_proxy::Call::reject_announcement { .. }) |
+					Call::Utility { .. } | Call::Multisig { .. }
+			),
+			ProxyType::Collator => matches!(
+				c,
+				Call::CollatorSelection { .. } | Call::Utility { .. } | Call::Multisig { .. }
+			),
+			ProxyType::Alliance => matches!(
+				c,
+				Call::AllianceMotion { .. } |
+					Call::Alliance { .. } |
+					Call::Utility { .. } | Call::Multisig { .. }
+			),
+		}
+	}
+	fn is_superset(&self, o: &Self) -> bool {
+		match (self, o) {
+			(x, y) if x == y => true,
+			(ProxyType::Any, _) => true,
+			(_, ProxyType::Any) => false,
+			(ProxyType::NonTransfer, _) => true,
```

Comment:
Why this NonTrarsfer line? It's not in statemint... I'm thinking it should be in both or neither.

Reply:
Because in Statemint, `NonTransfer` is [not a superset](https://github.com/paritytech/cumulus/blob/release-parachains-v9230/parachains/runtimes/assets/statemint/src/lib.rs#L351) to the other proxy types.

```rust
ProxyType::NonTransfer =>
	!matches!(c, Call::Balances { .. } | Call::Assets { .. } | Call::Uniques { .. }),
```

However, Statemint should indeed add a `(ProxyType::NonTransfer, ProxyType::CollatorSelection) => true,` entry.

Reply:
Ok, but should the line above be `(ProxyType::NonTransfer, o) => o != ProxyType::Any` for soundness of supertypes?
(I'm assuming we don't want NonTransfer and Any to both be supertypes of each other... that would feel to me like a footgun)

Reply:
But `match` goes in order and short circuits, right? So `(_, ProxyType::Any) => false,` would prevent any footguns w/r/t `Any`.

Reply:
It's complicated: https://www.reddit.com/r/rust/comments/pqnufj/rust_match_arm_priority/
As it turns out they do go in order because your using the guard clauses. Without the guard clauses it could optimise it in some fun ways.

Reply:
It must always use a fixed order when there is any `_` involved. There also doesn't seem to be any official documentation. There only is this reddit thread, where all are just guessing and don't have any information.

Reply:
I'm just going to come out with it: Basti's right. From the rust reference it's going to use top down ordering (or anything that optimises to an equivalent of that).